src/jit/lower.cpp

```diff
@@ -5207,7 +5207,7 @@ GenTree* Lowering::LowerArrElem(GenTree* node)
 
     for (unsigned char dim = 0; dim < rank; dim++)
     {
-        GenTree* indexNode = arrElem->gtArrElem.gtArrInds[dim];
+        GenTree* indexNode = arrElem->AsArrElem()->gtArrInds[dim];
```

Comment:
in that function `arrElem` is already `GenTreeArrElem*     arrElem = node->AsArrElem();` so you don't need these casts, just delete `->AsArrElem()`.